src/SheetsRegistry.js

```diff
@@ -1,12 +1,22 @@
 /* @flow */
-import type {ToCssOptions} from './types'
+import {createGenerateClassName} from './utils/generateClassName'
+import type {ToCssOptions, SheetsRegistryOptions} from './types'
 import type StyleSheet from './StyleSheet'
 
 /**
  * Sheets registry to access them all at one place.
  */
 export default class SheetsRegistry {
   registry: Array<StyleSheet> = []
+  registryOptions: SheetsRegistryOptions
+
+  constructor(options: SheetsRegistryOptions = {}) {
+    const generateClassName = options.generateClassName ||
+      (options.resetClassNameCounter === true && createGenerateClassName())
```

Comment:
Sorry about the slow reply. Somehow this notification slipped through the cracks. 

Anyways, that was my original plan, and I'm happy to go back to it if that's what folks want. My thinking had to do with backwards compatibility. Since the SheetsRegistry now has a classname generator in it, it's going to get passed into `createStyleSheet` by `react-jss`. When that happens, it's going to override any classname generator that was previously configured at the `jss` level, which would break existing code. I guess the question is: how does the registry's generator relate to generators configured in other places? The way I implemented it, it has the highest priority, but you have to turn it on. That way, folks that have configured a hashing generator in their jss instance don't see their code strangely break.

The more I think about it, I'm not satisfied with the approach I took. I think that perhaps the registry generator should get passed into `createStyleSheet` using a different `options` key, and not under `generateClassName`. Something like `options.registryGenerateClassName`. Maybe even `options.sheetRegistryContext`. Then `jss` will only use it if a generator hasn't already been configured somewhere else.